neqo-transport/src/connection.rs

```diff
@@ -1140,12 +1144,18 @@ impl Connection {
                     msg,
                     ..
                 } => {
-                    if epoch != 3 {
-                        continue;
-                    }
-
                     if self.flow_mgr.borrow().need_close_frame() {
-                        self.flow_mgr.borrow_mut().set_need_close_frame(false);
+                        if epoch == highest_state {
+                            self.flow_mgr.borrow_mut().set_need_close_frame(false);
```

Comment:
I think that this will interfere with key updates.  What we should instead do is send in every epoch and clear this flag at the end of the function if this function was called.

That is:

```rust
let mut close_sent = false;
// ... iterate
if self.flow_mgr.borrow().need_close_frame() {
    close_sent = true;
    //....
}
// Finally
if close_sent {
    self.flow_mgr.borrow_mut().set_close_needed(false);
}
```

Reply:
:+1: